### ISSUE TITLE
Rawhide: rpm build is broken: Too many levels of symbolic links

### ISSUE THREAD

**Issue by edsantiago** [containers/buildah#2264]
The last two builds (1.15.0-0 [.21](https://koji.fedoraproject.org/koji/buildinfo?buildID=1486746) and [.22](https://koji.fedoraproject.org/koji/buildinfo?buildID=1486920)) on rawhide have [crashed](https://kojipkgs.fedoraproject.org//work/tasks/1274/42921274/build.log):
```
RPM build errors:
    Recognition of file "/builddir/build/BUILDROOT/buildah-1.15.0-0.22.dev.git912ca5a.fc33.x86_64/usr/share/buildah/test/system/bud/dockerignore2/subdir/circular-link" failed: mode 120777 broken symbolic link to circular-link (Too many levels of symbolic links)
```
The file `tests/bud/dockerignore2/subdir/circular-link` was introduced in #1583, almost a year ago. The next obvious place to look is `rpm` itself, but looking through its [git log](https://github.com/rpm-software-management/rpm/commits/master) I see no smoking guns. *Maybe* [c6ff61448](https://github.com/rpm-software-management/rpm/commit/c6ff61448a4ead0540130c9a6eb052fcd71b2624) but I'm not smart enough to be sure, and March 5 seems too recent to actually be pushed out and affecting us already.

### Solution

First idea: git-remove the offending link, and create it in the test script itself. No-go, because if buildah-rootless ever works we will need to test it; and this directory is not writable by nonroot.

Next idea: git-remove the link, create it in an RPM %postinstall script. Nonstarter: it would not work in CI, and would have to be replicated in all distros that package tests. Plus, yuk.

Next idea: git-remove the link; move this test to a new subtest which `cp -r`s the majority of the `bud/dockerignore2` directory into a tempdir, then creates the circular-link. I think this is the only possibility but hope that others have better ideas.

**Comment by edsantiago** [comment]
Filed https://github.com/rpm-software-management/rpm/issues/1159

**Comment by edsantiago** [comment]
Fixed by #2268
